h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -702,16 +726,22 @@ protected ResultInterface queryWithoutCache(int maxRows, ResultTarget target) {
                 !session.getDatabase().getSettings().optimizeInsertFromSelect)) {
             result = createLocalResult(result);
         }
+        boolean quickOffset = true;
```

Comment:
maybe add comment here
  // Do not add rows before OFFSET to result if possible
since this Select class is getting quite complex, need all the code docs we can :-)